src/utils/pgwire/src/pg_server.rs

```diff
@@ -132,6 +138,12 @@ impl ExecContextGuard {
     }
 }
 
+impl Drop for ExecContext {
+    fn drop(&mut self) {
+        *self.last_idle_instant.lock() = Some(Instant::now());
```

Comment:
As the `last_idle_instant` is only updated when execution completes, if a batch query runs for a long time e.g. scanning a large amount of data, will it be killed?

Reply:
> As the `last_idle_instant` is only updated when execution completes, if a batch query runs for a long time e.g. scanning a large amount of data, will it be killed? 

According to the postgresql docs, this variable doesn't care about how long a statement is running, which should be handled by parameters like statement timeout

Reply:
Yes, we won't kill any query, but unpin the snapshot (either by background monitor or session termination triggered by a later statement). The session would be terminated when the later statement arrives.

Reply:
> but unpin the snapshot

By "killed" I mean any means that cause the query to end. When a snapshot is unpinned, it's possible that the iterator runs into errors and terminates, is it the cases?

Reply:
I just ensure there is no running SQL in the current session during calling `check_idle_in_transaction_timeout`, so we don't need to worry about that.